src/System.Console/src/System/ConsoleKeyInfo.cs

```diff
@@ -70,7 +70,11 @@ public bool Equals(ConsoleKeyInfo obj)
 
         public override int GetHashCode()
         {
-            return (int)_keyChar | (int)_mods;
+            // For all typical cases we can fit all bits losslessly into the hash code:
+            // _keyChar could be any 16-bit value (though is most commonly ASCII). Use 16 bits.
+            // _key can theoretically be 32-bit, but only has enum defined values up to 255.  Use 8 bits.
+            // _mods can also theoretically be 32-bit, but only has enum defined values up to 4. Use 8 bits.
+            return _keyChar | ((int)_key << 16) | ((int)_mods << 24);
```

Comment:
The comment doesn't match the code - the comment says `Use 8 bits`, so fix that for sure. But you're right, you're using 16 bits of `_key` with a potential collision between infrequently used bits of `_key` and frequently used bits of `_mods`.

Given that you are actually using more than 8 bits from `_key`, it probably doesn't matter about potential overlap with `_mods` and using byte aligned shifting might result in better code, so maybe just fix the comment.